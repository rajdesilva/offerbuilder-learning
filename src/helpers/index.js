export * from './api-config';
export * from './history';
