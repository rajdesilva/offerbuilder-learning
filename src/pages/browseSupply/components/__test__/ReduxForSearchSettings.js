export const ReduxForSearchSettings = {
  searchparams: {
    destination: {},
    los: 1,
    distance: 10,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [{}],
      },
    ],
    dateRange: {
      startDate: "05/17/2021",
      endDate: "07/17/2021",
    },

    lcn: false,
    remainingCapitalPool: "",
    sortBy: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "PCT",
        name: "PCT",
      },
      {
        id: "PCT",
        name: "ATG",
      },
      {
        id: "DEMO_OFFERBUILDER",
        name: "DEMO_OFFERBUILDER",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
