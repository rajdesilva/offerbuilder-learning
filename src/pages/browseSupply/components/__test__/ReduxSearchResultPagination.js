export const PaginationStore = {
  searchedproperties: {
    loading: false,
    properties: [{}],
    totalMatch: 15,
  },
  searchparams: {
    pageOffset: 0,
    pageSize: 10,
  },
};
