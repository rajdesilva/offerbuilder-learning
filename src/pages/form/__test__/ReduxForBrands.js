export const ReduxForBrands = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
        id: "idb",
        name: "Internal Demo Brand",
      },
    ],
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
