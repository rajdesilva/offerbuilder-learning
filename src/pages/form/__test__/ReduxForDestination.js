export const reduxStore = {
  supplysearchParams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [],
      channels: [],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channels: {
    // loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: null,
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "AT_KINDERHOTELS",
        name: "Ihr Bett im Allg√§u",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
