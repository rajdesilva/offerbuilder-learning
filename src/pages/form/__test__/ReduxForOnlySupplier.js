export const ReduxForOnlySupplier = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [
        {
          id: "ntp",
          name: "NTP",
        },
      ],
      channels: [
        {
          id: "AT_CORDIAL",
          name: "",
        },
      ],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
