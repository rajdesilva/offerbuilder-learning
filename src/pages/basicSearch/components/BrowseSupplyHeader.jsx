import React, { Fragment } from "react";
import { Button, Row, Col, Divider, Tooltip } from "antd";
import { Link, useRouteMatch } from "react-router-dom";
import { useSelector } from "react-redux";
import { FormattedMessage } from "react-intl";
import { checkIfUserHasRole } from "../../../helpers/utility";
import { appConstants } from "../../../common";
import styles from './css/BrowseSupplyHeader.module.less';

export function BrowseSupplyHeader() {
  const isBrowseSupplyPage = useRouteMatch({
    path: "/browse-supply/:page",
    strict: true,
    sensitive: true,
  });
  const onlySupplier = useSelector((state) =>
    window.getValue(state, "searchparams.onlySupplier")
  );

  return (
    <React.Fragment>
      <Row justify="space-between" className={styles.bsHeader}>
        <Col>
          <h2>
            <FormattedMessage id="nemo.global.search.of.all.properties" />
          </h2>
        </Col>
        {process.env.REACT_APP_HIDE_CLIENT_AND_STORE ? (
          <Fragment />
        ) : isBrowseSupplyPage &&
          checkIfUserHasRole(appConstants.USER_ROLE.ADMIN) ? (
          <Col>
            <Tooltip
              placement="left"
              title={<FormattedMessage id="nemo.addBrandsToCreateOffer" />}
              visible={onlySupplier}
              data-testid="add-brand-tooltip"
            >
              <Button
                type="primary"
                size="large"
                disabled={onlySupplier}
                data-testid="create-offer-search-results-btn"
              >
                <Link
                  to={{
                    pathname: "/offers/create-new-offer/2",
                    search: `${window.location.search}`,
                  }}
                >
                  <FormattedMessage id="nemo.createOfferWithSearchResult" />
                </Link>
              </Button>
            </Tooltip>
          </Col>
        ) : (
          <Fragment />
        )}
      </Row>
      {isBrowseSupplyPage ? <Divider className={styles.divider} /> : <Fragment />}
    </React.Fragment>
  );
}
