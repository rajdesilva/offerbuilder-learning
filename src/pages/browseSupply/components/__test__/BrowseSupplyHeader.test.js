import React from "react";
import { createStore } from "redux";
import { render, cleanup } from "../../../../helpers/testUtils";
import { BrowseSupplyHeader } from "../BrowseSupplyHeader";
import { MemoryRouter } from "react-router-dom";
import { tempStore } from "../../../offers/components/__test__/tempStore";
import * as Util from "../../../../helpers/utility/checkIfUserHasRole";

afterEach(cleanup);
describe("BrowseSupplyHeader component test", () => {
  test("create BrowseSupplyHeader components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/search-results"]}>
        <BrowseSupplyHeader />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;

    expect(container.firstChild).toMatchSnapshot();
  });
  test("BrowseSupplyHeader - Admin Role and supplier flag is off", () => {
    const store = createStore(() => ({ ...tempStore }));
    const checkUserPermission = jest.spyOn(Util, "checkIfUserHasRole");
    checkUserPermission.mockReturnValue(true);

    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/search-results"]}>
        <BrowseSupplyHeader />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { getByText, getByTestId } = wrapper;

    expect(getByText("Create Offer With Search Result")).toBeInTheDocument();
    expect(getByTestId("create-offer-search-results-btn")).not.toBeDisabled();
  });
  test("BrowseSupplyHeader - Admin Role and supplier flag is on", () => {
    const ts = tempStore;
    ts.searchparams.onlySupplier = true;
    const store = createStore(() => ({ ...ts }));
    const checkUserPermission = jest.spyOn(Util, "checkIfUserHasRole");
    checkUserPermission.mockReturnValue(true);

    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/search-results"]}>
        <BrowseSupplyHeader />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { getByText, getByTestId } = wrapper;

    expect(getByText("Create Offer With Search Result")).toBeInTheDocument();
    expect(getByTestId("create-offer-search-results-btn")).toBeDisabled();
  });
});
