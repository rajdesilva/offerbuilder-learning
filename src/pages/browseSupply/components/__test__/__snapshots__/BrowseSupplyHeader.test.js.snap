// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseSupplyHeader component test create BrowseSupplyHeader components snapshot 1`] = `
<div
  class="ant-row ant-row-space-between"
>
  <div
    class="ant-col"
  >
    <h2>
      Browse Supply
    </h2>
  </div>
</div>
`;
